core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/FileAttachment.java

```diff
@@ -0,0 +1,109 @@
+package uk.ac.bbsrc.tgac.miso.core.data.impl;
+
+import java.io.Serializable;
+import java.util.Date;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+
+import com.eaglegenomics.simlims.core.User;
+
+@Entity(name = "Attachment")
+public class FileAttachment implements Serializable {
+
+  private static final long serialVersionUID = 1L;
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private long attachmentId;
+  private String filename;
+  private String path;
+
+  @ManyToOne(targetEntity = UserImpl.class)
+  @JoinColumn(name = "creator", nullable = false)
+  private User creator;
+
+  @Temporal(TemporalType.TIMESTAMP)
+  private Date created;
+
+  public long getId() {
+    return attachmentId;
+  }
+
+  public void setId(long attachmentId) {
+    this.attachmentId = attachmentId;
+  }
+
+  public String getFilename() {
+    return filename;
+  }
+
+  public void setFilename(String filename) {
+    this.filename = filename;
+  }
+
+  public String getPath() {
+    return path;
+  }
+
+  public void setPath(String path) {
+    this.path = path;
+  }
+
+  public User getCreator() {
+    return creator;
+  }
+
+  public void setCreator(User creator) {
+    this.creator = creator;
+  }
+
+  public Date getCreationTime() {
+    return created;
+  }
+
+  public void setCreationTime(Date creationTime) {
+    this.created = creationTime;
+  }
+
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = 1;
+    result = prime * result + (int) (attachmentId ^ (attachmentId >>> 32));
+    result = prime * result + ((created == null) ? 0 : created.hashCode());
+    result = prime * result + ((creator == null) ? 0 : creator.hashCode());
+    result = prime * result + ((filename == null) ? 0 : filename.hashCode());
+    result = prime * result + ((path == null) ? 0 : path.hashCode());
+    return result;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 20 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)